### DISCUSSION TITLE
How to create a CarePlan for a location without a patient

### DISCUSSION THREAD

**Discussion by rkareko** [opensrp/fhircore#320] [Ideas]
How to create a CarePlan for a location without a patient

@rkodev @ekigamba @f-odhiambo @ndegwamartin

**Comment by f-odhiambo** [comment]
More on the CarePlan generation is discussed [here](https://github.com/opensrp/fhircore/discussions/309) 

_**The sort answer Yes**_ 

From the preliminary research, the CarePlan generation can take two approaches 
1. Manual approach  - Not advised
2. Automated approach using the  $apply operation  - run the operation $apply with params $appy(subject) = CarePlan for a patient. The subject(s) that is/are the target of the plan to be applied. The subject may be a Patient, Practitioner, Organization, Location, Device, or Group. Subjects provided in this parameter will be resolved as the subject of the PlanDefinition based on the type of the subject. If multiple subjects of the same type are provided, the behaviour is implementation-defined
 
The result of this operation is a CarePlan resource with a single activity represented by a RequestGroup. The RequestGroup will have actions for each of the applicable actions in the plan based on evaluating the applicability condition in context. More on this operation as currently implemented [here](http://build.fhir.org/ig/HL7/cqf-recommendations/OperationDefinition-cpg-plandefinition-apply.html)  


cc @Rkareko
